sql/core/src/test/scala/org/apache/spark/sql/execution/command/v1/ShowTablesSuite.scala

```diff
@@ -165,4 +154,222 @@ class ShowTablesSuite extends ShowTablesSuiteBase with CommandSuiteBase {
       }
     }
   }
+
+  test("show table extended in non-partitioned table") {
+    val namespace = "ns1"
+    val table = "tbl"
+    withNamespaceAndTable(namespace, table, catalog) { tbl =>
+      sql(s"CREATE TABLE $tbl (id bigint, data string) $defaultUsing")
+      val e = intercept[AnalysisException] {
+        sql(s"SHOW TABLE EXTENDED IN $catalog.$namespace LIKE '$table' PARTITION(id = 1)")
+      }
+      checkError(
+        exception = e,
+        errorClass = "_LEGACY_ERROR_TEMP_1251",
+        parameters = Map("action" -> "SHOW TABLE EXTENDED", "tableName" -> table)
+      )
+    }
+  }
+
+  test("show table extended in multi partition key - " +
```

Comment:
what's the v2 difference for this test?

Reply:
The difference between the results of `v1` and `v2` is not significant, mainly due to the addition of information in the red box below in the `hive` results
<img width="614" alt="image" src="https://github.com/apache/spark/assets/15246973/046b2032-cd0b-4a23-9d82-5b87c42774c5">

Reply:
can we have some way to abstract it out? e.g. the sub test suite can add some custom information in the result.

Reply:
Okay, I'll give it a try.

Reply:
Done.
I have updated some logic of UT. Previously, I used the `exclude method` to exclude fields with `non-deterministic `values, but now I have changed it to `replace` with `a fixed template value` for better comparison of results. At present, the difference displayed by UT should be relatively clear. eg:

`Created Time: Wed Nov 01 04:51:19 PDT 2023` => `Created Time: <created time>`
`Last Access: UNKNOWN` => `Last Access: <last access>`
`Created By: Spark` => `Created By: <created by>`